rust/scannerctl/src/interpret/mod.rs

```diff
@@ -193,7 +213,15 @@ pub fn run(
         }
         (Db::InMemory, Some(path)) => {
             let storage = DefaultDispatcher::new(true);
-            let builder = RunBuilder::default().loader(create_fp_loader(&storage, path)?);
+
+            let is_json = path.extension().map(|ext| ext == "json").unwrap_or(false);
```

Comment:
seems convoluted, going to provide an example later.

Reply:
convoluted was a bit dramatic, but I think:
```
            let is_json = path.extension().filter(|ext| *ext == "json").is_some();
```
is a bit easier to read, even when inlined in the if statement instead of a variable.